GameObject.hpp

```diff
@@ -193,6 +193,36 @@ namespace spic {
              */
             const spic::Transform& Transform() const;
 
+            /**
+             * The parent of this GameObject.
+             * @return A weak pointer to the parent.
+             */
+            std::weak_ptr<GameObject> Parent();
+
+            /**
+             * The parent of this GameObject.
+             * @param parent A weak pointer to the new parent
+             */
+            void Parent(std::weak_ptr<GameObject> parent);
+
+            /**
+             * Returns a list of children in this GameObject.
+             * @return A list of shared pointers to the children.
+             */
+            std::vector<std::shared_ptr<GameObject>> Children() const;
```

Comment:
Deze vector moet een reference zijn, en kan waarschijnlijk ook const zijn

Reply:
Dan kun je er net zo goed een non-const reference van maken, zodat de AddChild en RemoveChild niet nodig zijn.

Reply:
Daar ben ik het niet mee eens, maak er op zn minst een normale reference van, want anders doet ie steeds die vector copyen

Reply:
Dan maak ik er een const reference van, zodat het alleen via de Add/Remove gedaan kan worden zodat daar nog eventuele logica in de methodes kunnen